src/ape/api/providers.py

```diff
@@ -382,6 +382,23 @@ class ProviderAPI:
     request_header: str
     """A header to set on HTTP/RPC requests."""
 
+    def __post_init__(self):
+        abstract_test_api_methods = set(TestProviderAPI.__abstractmethods__) - set(
+            ProviderAPI.__abstractmethods__
+        )
+
+        def raise_not_implemented(method_name: str):
+            raise NotImplementedError(
+                f"Propvider '{self.name}' is not a 'TestProviderAPI' "
+                f"and thus method '{method_name}' is not implemented."
+            )
+
+        for test_api_method in abstract_test_api_methods:
+            try:
+                getattr(self, test_api_method) is None
+            except AttributeError:
+                setattr(self, test_api_method, lambda: raise_not_implemented(test_api_method))
+
```

Comment:
Couldn't this all be done by adding it to the class externally? Then we don't need `__post_init__`

Reply:
Sure. `setattr` needs an instance to work though, and it is a bit weird to have code for a class happening in some other class.

Reply:
Lol, I just realized I don't need to use `setattr` and then we can probably do this at the module level, I like that

Reply:
I can just duplicate the methods in both if you want.